### DISCUSSION TITLE
Packaging relation_extractor - ValueError: Existing factory.

### DISCUSSION THREAD

**Discussion by biagiodistefano** [explosion/spaCy#10706] [Help: Other Questions] [answered]
Hi!

I have successfully trained my custom ner + relation_extractor starting from [this great tutorial](https://github.com/explosion/projects/tree/v3/tutorials/rel_component).

I tried to package it using

```
python -m spacy package training/model-best packaged --code scripts/rel_pipe.py,scripts/rel_model.py
```

So far, so good.

But when I pip install the resulting `*.tar.gz` and import it with `spacy.load("my_model"), first I get the following error:

```
...
from . import rel_pipe
  File "/Users/biagio/repos/my-project/venv/lib/python3.10/site-packages/it_my_model/rel_pipe.py", line 17, in <module>
    Doc.set_extension("rel", default={})
  File "spacy/tokens/doc.pyx", line 138, in spacy.tokens.doc.Doc.set_extension
ValueError: [E090] Extension 'rel' already exists on Doc. To overwrite the existing extension, set `force=True` on `Doc.set_extension`.
```

So first I tried with adding `force=True` as suggested by the stack. However, then this follows:

```
...
from . import rel_pipe
  File "/Users/biagio/repos/my_project/venv/lib/python3.10/site-packages/it_my_model/rel_pipe.py", line 39, in <module>
    def make_relation_extractor(
  File "/Users/biagio/repos/my_project/venv/lib/python3.10/site-packages/spacy/language.py", line 486, in add_factory
    raise ValueError(err)
ValueError: [E004] Can't set up pipeline component: a factory for 'relation_extractor' already exists. Existing factory: <function make_relation_extractor at 0x1164c0310>. New factory: <function make_relation_extractor at 0x1164c2c20>
```

Any advice on how to fix this? Thanks.

**Comment by thomashacker** [comment, marked as answer]
Hello,
if you installed your model via `pip`, why are you still importing it locally? `from . import rel_pipe`
I think this might be the culprit here 😄

**Reply by biagiodistefano** [reply]
Oh my, thanks! I just forgot to delete the import from when I was developing the thing 😅 I spent so many hours in this for nothing. Thanks!
